sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/history/OfferOutcomeTracker.java

```diff
@@ -0,0 +1,99 @@
+package com.mesosphere.sdk.offer.history;
+
+import com.google.common.collect.EvictingQueue;
+import j2html.tags.DomContent;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.util.Strings;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import static j2html.TagCreator.*;
+
+import java.sql.Timestamp;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class OfferOutcomeTracker {
+    private final EvictingQueue<OfferOutcome> outcomes;
+    private static final int DEFAULT_CAPACITY = 100;
+
+    public OfferOutcomeTracker() {
+        this.outcomes = EvictingQueue.create(DEFAULT_CAPACITY);
+    }
+
+    public OfferOutcomeTracker(int capacity) {
+        this.outcomes = EvictingQueue.create(capacity);
+    }
+
+    public void track(OfferOutcome outcome) {
+        outcomes.add(outcome);
+    }
+
+    public void track(OfferOutcome... outcomes) {
+        Arrays.stream(outcomes).forEach(this::track);
```

Comment:
Is there any advantage to calling `this.outcomes.addAll()` here directly?

Reply:
That would be better. Swapping it for `this.outcomes.addAll(Arrays.asList(outcomes));`

Reply:
(I wish I knew enough Java to know which of these is "faster"